packages/flutter/lib/src/rendering/sliver_fill.dart

```diff
@@ -142,61 +142,55 @@ class RenderSliverFillRemaining extends RenderSliverSingleBoxAdapter {
 
   @override
   void performLayout() {
-    double childExtent;
-    double extent = constraints.viewportMainAxisExtent - constraints.precedingScrollExtent;
-    double maxExtent = constraints.remainingPaintExtent - math.min(constraints.overlap, 0.0);
-
-    if (hasScrollBody) {
-      extent = maxExtent;
-      if (child != null)
-        child.layout(
-          constraints.asBoxConstraints(
-            minExtent: extent,
-            maxExtent: extent,
-          ),
-          parentUsesSize: true,
-        );
-    } else if (child != null) {
+    final bool previousSliversCanFillViewport = constraints.viewportMainAxisExtent < constraints.precedingScrollExtent;
+    final double availableExtent = constraints.remainingPaintExtent - math.min(constraints.overlap, 0.0);
+
+    double childExtent = availableExtent;
+    if (child != null) {
+      double minExtent = 0.0;
+      double maxExtent = double.infinity;
+      if (hasScrollBody) {
+        minExtent = availableExtent;
+        maxExtent = availableExtent;
```

Comment:
if hasScrollBody && !fillOverscroll, shouldn't it have min = `constraints.viewportMainAxisExtent - constraints.precedingScrollExtent` ?

Reply:
Not according to the old code :) Maybe that's a bug there? @Piinks

Reply:
FWIW, all tests still pass with this suggested change.

Reply:
IIRC, the introduction of `hasScrollBody` was to maintain the behavior that existed before this was refactored, which `NestedScrollView` is dependent upon. `hasScrollBody: true` makes `fillOverscroll` irrelevant, which is documented above.

Reply:
Seems like these flags should be orthogonal, otherwise it should be one flag that's an enum.